KHJ/6Day/예산.js

```diff
@@ -0,0 +1,13 @@
+function solution(d, budget) {
+    d.sort((a, b) => a - b);
+    let totalTeam = 0;
+    for(let i = 0 ; i < d.length; i++){
+        if (budget - d[i] >= 0) {
+            budget -= d[i];  //예산에서 현재 요청비용 빼기
+            totalTeam++; //if문이 돌아가서 totalTeam 하나 증가
+        }else{
+            break;
+        }
+    } return totalTeam;
+}
```

Comment:
```suggestion
function solution(d, budget) {
    let totalTeam = 0;
    
    d.sort((a, b) => a - b);
    
    for (let i = 0 ; i < d.length; i++ ) {
        if (budget - d[i] >= 0) {
            budget -= d[i];  // 예산에서 현재 요청비용 빼기
            totalTeam++; // if문이 돌아가서 totalTeam 하나 증가
        }
    } 
    return totalTeam;
}
```
@eric5025 
- 웬만하면 변수 선언이 로직의 상단으로 올라와야 합니다.
- 공백, 줄바꿈이 아직 컨벤션을 지켜지지 못하는 부분이 많습니다. 
    - 이 부분 개선이 잘안되는데 어떤 이유가 있을까요? 🤔
- 현재 if else문의 else문은 불 필요해 보여요! 제거 할 수 있을 것 같습니다
- `sort`, `splice`는 대표적으로 원본 배열을 변경하는 `사이드 이펙트`를 발생하는 배열 메서드입니다.
```ts
const arr = [4, 6, 1, 2, 3];
arr.sort((a, b) => a - b);

// 원본 배열이 변경!
console.log(arr); // [1, 2, 3, 4, 5] 
```
- 원본 배열을 바꾸는 행위가 과연 안정성있는 코드 작성 방법일까요? 한번 고민해보시고 아래 코멘트도 확인해주세요 :)
    - https://github.com/modern-agile-team/9term-assignment/pull/28#discussion_r1873250608

Reply:
넵 https://github.com/modern-agile-team/9term-assignment/pull/40#discussion_r1878293711 이 코멘트를 보면서 스프레드 연산자로 얕은복사를 이해래서 활용해보도록하겠습니다. 또한 공백 줄바꿈은 제가 좀더 신경쓰겠습니다
지적해주셔서 감사합니다.